### ISSUE TITLE
robot.PlanToNamedConfiguration( 'home', timeout = 0) fails

### ISSUE THREAD

**Issue by garthz** [personalrobotics/herbpy#12]
The expected behavior is that the function runs asynchronously, but instead it now fails with a `PlanningError`.

Test:
```python
$ rosrun herbpy console.py --sim
In [1]: robot.PlanToNamedConfiguration('home', timeout=0)
```

Without the timeout parameter, this succeeds.  With it, it produces the following error stream.
Note the complaint from `[OMPLSimplifier.cpp:109 PlanPath]`; is it using the wrong argument?

Error output follows:

```
openravepy.inversekinematics: GetDefaultIndices, found 3-intersection centered on index 6
[INFO] [openravepy.inversekinematics:inversekinematics.py:449]:GetDefaultIndices: found 3-intersection centered on index 6
openravepy.inversekinematics: GetDefaultIndices, found 3-intersection centered on index 11
[INFO] [openravepy.inversekinematics:inversekinematics.py:449]:GetDefaultIndices: found 3-intersection centered on index 11
openravepy.inversekinematics: GetDefaultIndices, found 3-intersection centered on index 17
[INFO] [openravepy.inversekinematics:inversekinematics.py:449]:GetDefaultIndices: found 3-intersection centered on index 17
openravepy.inversekinematics: GetDefaultIndices, found 3-intersection centered on index 22
[INFO] [openravepy.inversekinematics:inversekinematics.py:449]:GetDefaultIndices: found 3-intersection centered on index 22
openravepy.inversekinematics: GetDefaultIndices, found 3-intersection centered on index 6
[INFO] [openravepy.inversekinematics:inversekinematics.py:449]:GetDefaultIndices: found 3-intersection centered on index 6
openravepy.inversekinematics: GetDefaultIndices, found 3-intersection centered on index 11
[INFO] [openravepy.inversekinematics:inversekinematics.py:449]:GetDefaultIndices: found 3-intersection centered on index 11
openravepy.inversekinematics: GetDefaultIndices, found 3-intersection centered on index 17
[INFO] [openravepy.inversekinematics:inversekinematics.py:449]:GetDefaultIndices: found 3-intersection centered on index 17
openravepy.inversekinematics: GetDefaultIndices, found 3-intersection centered on index 22
[INFO] [openravepy.inversekinematics:inversekinematics.py:449]:GetDefaultIndices: found 3-intersection centered on index 22
/homes/garthz/ln/local/ros-hydro/src/prpy/src/prpy/planning/snap.py:108: FutureWarning: comparison to `None` will result in an elementwise object comparison in the future.
  params.SetGoalConfig(goal)
/homes/garthz/ln/local/ros-hydro/src/prpy/src/prpy/planning/snap.py:109: FutureWarning: comparison to `None` will result in an elementwise object comparison in the future.
  check = params.CheckPathAllConstraints(curr, goal, [], [], 0., Closed)
/homes/garthz/ln/local/ros-hydro/src/prpy/src/prpy/planning/snap.py:125: FutureWarning: comparison to `None` will result in an elementwise object comparison in the future.
  active_indices, False)
/homes/garthz/ln/local/ros-hydro/src/prpy/src/prpy/planning/snap.py:127: FutureWarning: comparison to `None` will result in an elementwise object comparison in the future.
  active_indices, False)
/homes/garthz/ln/local/ros-hydro/src/prpy/src/prpy/planning/snap.py:130: FutureWarning: comparison to `None` will result in an elementwise object comparison in the future.
  traj.Insert(0, waypoints.ravel())
openravepy.inversekinematics: GetDefaultIndices, found 3-intersection centered on index 6
[INFO] [openravepy.inversekinematics:inversekinematics.py:449]:GetDefaultIndices: found 3-intersection centered on index 6
openravepy.inversekinematics: GetDefaultIndices, found 3-intersection centered on index 11
[INFO] [openravepy.inversekinematics:inversekinematics.py:449]:GetDefaultIndices: found 3-intersection centered on index 11
openravepy.inversekinematics: GetDefaultIndices, found 3-intersection centered on index 17
[INFO] [openravepy.inversekinematics:inversekinematics.py:449]:GetDefaultIndices: found 3-intersection centered on index 17
openravepy.inversekinematics: GetDefaultIndices, found 3-intersection centered on index 22
[INFO] [openravepy.inversekinematics:inversekinematics.py:449]:GetDefaultIndices: found 3-intersection centered on index 22
/homes/garthz/ln/local/ros-hydro/src/prpy/src/prpy/base/robot.py:170: FutureWarning: comparison to `None` will result in an elementwise object comparison in the future.
  cloned_env.Cloned(self).SetActiveDOFs(traj_dofs)
openravepy.inversekinematics: GetDefaultIndices, found 3-intersection centered on index 6
[INFO] [openravepy.inversekinematics:inversekinematics.py:449]:GetDefaultIndices: found 3-intersection centered on index 6
openravepy.inversekinematics: GetDefaultIndices, found 3-intersection centered on index 11
[INFO] [openravepy.inversekinematics:inversekinematics.py:449]:GetDefaultIndices: found 3-intersection centered on index 11
openravepy.inversekinematics: GetDefaultIndices, found 3-intersection centered on index 17
[INFO] [openravepy.inversekinematics:inversekinematics.py:449]:GetDefaultIndices: found 3-intersection centered on index 17
openravepy.inversekinematics: GetDefaultIndices, found 3-intersection centered on index 22
[INFO] [openravepy.inversekinematics:inversekinematics.py:449]:GetDefaultIndices: found 3-intersection centered on index 22
[OMPLSimplifier.cpp:109 PlanPath] Time limit must be positive; got 0.000000.---------------------------------------------------------------------------
PlanningError                             Traceback (most recent call last)
/homes/garthz/ln/local/ros-hydro/src/herbpy/scripts/console.py in <module>()
----> 1 robot.PlanToNamedConfiguration('home', timeout=0)

/homes/garthz/ln/local/ros-hydro/src/prpy/src/prpy/base/robot.pyc in wrapper_method(*args, **kw_args)
     94             @functools.wraps(delegate_method)
     95             def wrapper_method(*args, **kw_args):
---> 96                 return self._PlanWrapper(delegate_method, args, kw_args)
     97 
     98             return wrapper_method

/homes/garthz/ln/local/ros-hydro/src/prpy/src/prpy/base/robot.pyc in _PlanWrapper(self, planning_method, args, kw_args)
    338             # Optionally execute the trajectory.

    339             if kw_args.get('execute', True):
--> 340                 result = self.ExecutePath(result, **kw_args)
    341 
    342             return result

/homes/garthz/ln/local/ros-hydro/src/prpy/src/prpy/base/robot.pyc in ExecutePath(self, path, simplify, smooth, defer, timeout, **kwargs)
    197         else:
    198             return do_execute(path, simplify=simplify, smooth=smooth,
--> 199                               timeout=timeout, **kwargs)
    200 
    201     def ExecuteTrajectory(self, traj, defer=False, timeout=None, period=0.01, **kw_args):

/homes/garthz/ln/local/ros-hydro/src/prpy/src/prpy/base/robot.pyc in do_execute(path, simplify, smooth, timeout, **kwargs)
    173                 if simplify:
    174                     path = self.simplifier.ShortcutPath(cloned_robot, path, defer=False,
--> 175                                                         timeout=timeout, **kwargs)
    176 
    177                 retimer = self.smoother if smooth else self.retimer

/homes/garthz/ln/local/ros-hydro/src/prpy/src/prpy/planning/base.pyc in __call__(self, instance, robot, *args, **kw_args)
     81                 return wrap_future(executor.submit(call_planner))
     82             else:
---> 83                 return call_planner()
     84 
     85     def __get__(self, instance, instancetype):

/homes/garthz/ln/local/ros-hydro/src/prpy/src/prpy/planning/base.pyc in call_planner()
     70                 try:
     71                     planner_traj = self.func(instance, cloned_robot,
---> 72                                              *args, **kw_args)
     73                     return CopyTrajectory(planner_traj, env=env)
     74                 finally:

/homes/garthz/ln/local/ros-hydro/src/prpy/src/prpy/planning/ompl.pyc in ShortcutPath(self, robot, path, timeout, **kwargs)
    213                            PlannerStatus.InterruptedWithSolution ]:
    214             raise PlanningError('Simplifier returned with status {0:s}.'.format(
--> 215                                 str(status)))
    216 
    217         return output_path

PlanningError: Simplifier returned with status Failed.
```

**Comment by mkoval** [comment]
Yes, it looks like `OMPLSimplifier` [uses the `timeout` argument](https://github.com/personalrobotics/prpy/blob/master/src/prpy/planning/ompl.py#L196) as the planning time limit. We use the `timelimit` keyword argument for this in all of our other planners. As a temporary workaround, you can manually call the simplifier and retimer:

```python
path = robot.PlanToNamedConfiguration('home', execute=False)
simplified_path = robot.simplifier.ShortcutPath(robot, path)
traj = robot.smoother.RetimeTrajectory(robot, simplified_path)
robot.ExecuteTrajectory(traj, timeout=0)
```

We are working on another refactor to the simplification/smoothing/retiming pipeline that should further simplify this process. This raises some a deeper issue that we need to address in that refactor, see https://github.com/personalrobotics/prpy/issues/73.

**Comment by mkoval** [comment]
This was fixed in personalrobotics/prpy#73 and now works as expected.
